### DISCUSSION TITLE
Is there anyway to get filelist from q-input(file type)?

### DISCUSSION THREAD

**Discussion by CodingSnorlax** [quasarframework/quasar#13796] [CLI - PWA mode] [answered]
Hi, everyone. I have a question about a file type q-input.
I'm trying to get the [FileList](https://developer.mozilla.org/en-US/docs/Web/API/FileList) from the input, and I need to render each filelist's name on my template, but it didn't seem to work.
I'm not sure about why?
This is my[ codepen](https://codepen.io/Coding_Snorlax/pen/JjLPMNR), thanks a lot!

**Comment by MilosPaunovic** [comment, marked as answer]
Add `v-model` to input.

**Reply by CodingSnorlax** [reply]
sorry, i'm not sure about what you mean.
i have read the [document](https://quasar.dev/vue-components/input#input-of-file-type), but it suggests that not to use v-model on q-input with a file type?

**Reply by MilosPaunovic** [reply]
Is there a project requirement which stops you from using `QFile`? It might make thigs easier. Or, you could go the way you've started, listen to the `@update:model-value` event and use the files provided as argument to save them (or only the names) to another variable.

**Reply by CodingSnorlax** [reply]
Thanks for your advice!  
I just took your advice and set a watch property on the filelist data, and then i can get the file name!

Yeah, I'm working on a project that needs to customize a file upload component, and it's a little complicated.
Besides the file name, i'm going to use [filereader](https://developer.mozilla.org/en-US/docs/Web/API/FileReader) to get image url and render the image, when the uploaded file is an image type.

Thanks again. Really appreciate your help!!!
